Redundant imports in `System.File.Platform` with filepath >= 1.5

```
libraries/file-io/posix/System/File/Platform.hs:30:1: error: [GHC-66111] [-Wunused-imports, Werror=unused-imports]
    The import of ‘System.OsString.Internal.Types’ is redundant
      except perhaps to import instances from ‘System.OsString.Internal.Types’
    To import instances alone, use: import System.OsString.Internal.Types()
   |
30 | import "os-string" System.OsString.Internal.Types (PosixString(..), PosixChar(..))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

libraries/file-io/posix/System/File/Platform.hs:31:1: error: [GHC-66111] [-Wunused-imports, Werror=unused-imports]
    The qualified import of ‘System.OsString.Data.ByteString.Short’ is redundant
      except perhaps to import instances from ‘System.OsString.Data.ByteString.Short’
    To import instances alone, use: import System.OsString.Data.ByteString.Short()
   |
31 | import qualified "os-string" System.OsString.Data.ByteString.Short as BC
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Revision history for file-io
 
+## 0.1.5 -- 20XX-XX-XX
+
+* Remove redundant imports
+
 ## 0.1.4 -- 2024-08-10
 
 * fix build on GHC <= 8.10

--- a/file-io.cabal
+++ b/file-io.cabal
@@ -1,6 +1,6 @@
 cabal-version:      2.4
 name:               file-io
-version:            0.1.4
+version:            0.1.5
 synopsis:           Basic file IO operations via 'OsPath'
 description:        Basic file IO operations like Prelude, but for 'OsPath'.
 homepage:           https://github.com/hasufell/file-io

--- a/posix/System/File/Platform.hs
+++ b/posix/System/File/Platform.hs
@@ -26,10 +26,7 @@ import System.Posix.Internals (c_getpid)
 import GHC.IORef (atomicModifyIORef'_)
 import Foreign.C (getErrno, eEXIST, errnoToIOError)
 
-#if MIN_VERSION_filepath(1, 5, 0)
-import "os-string" System.OsString.Internal.Types (PosixString(..), PosixChar(..))
-import qualified "os-string" System.OsString.Data.ByteString.Short as BC
-#else
+#if !MIN_VERSION_filepath(1, 5, 0)
 import Data.Coerce (coerce)
 import "filepath" System.OsString.Internal.Types (PosixString(..), PosixChar(..))
 import qualified "filepath" System.OsPath.Data.ByteString.Short as BC

